{
  "sha": "5beafce944b6b200d1fa80d7f4a8656cb2ecf2fa",
  "node_id": "MDY6Q29tbWl0MTM4Njg2ODE6NWJlYWZjZTk0NGI2YjIwMGQxZmE4MGQ3ZjRhODY1NmNiMmVjZjJmYQ==",
  "commit": {
    "author": {
      "name": "Tom de Vries",
      "email": "tdevries@suse.de",
      "date": "2019-07-29T13:07:47Z"
    },
    "committer": {
      "name": "Tom de Vries",
      "email": "tdevries@suse.de",
      "date": "2019-07-29T13:07:47Z"
    },
    "message": "[gdb/testsuite] Fix python.exp with check-read1\n\nwhen running python/python.exp with check-read1, we get:\n...\n(gdb) PASS: gdb.python/python.exp: prompt substitution readline - end\npython gdb.prompt_hook = error_prompt^M\nPython Exception <type 'exceptions.RuntimeError'> Python exception calledPASS: gdb.python/python.exp: set hook\n: ^M\n(gdb) PASS: gdb.python/python.exp: set the hook to default\npython gdb.prompt_hook = None^M\n(gdb) PASS: gdb.python/python.exp: set print-stack full for prompt error test\nset python print-stack full^M\n(gdb) FAIL: gdb.python/python.exp: set the hook\npython gdb.prompt_hook = error_prompt^M\nTraceback (most recent call last):^M\n  File \"<string>\", line 3, in error_prompt^M\nRuntimeError: Python exception called^M\n(gdb) FAIL: gdb.python/python.exp: set the hook to default\n...\n\nThe problem is that gdb_test_multiple here:\n...\ngdb_test_multiple \"python gdb.prompt_hook = error_prompt\" \"set the hook\" {\n    -re \"Python Exception (exceptions.RuntimeError|<(type 'exceptions.|class ')RuntimeError'>) Python excepti\non called.*\" {\n       pass \"set hook\"\n    }\n}\n...\nspecifies a regexp that ends with \".*\" but doesn't specify the expected\n$gdb_prompt.\n\nConsequently, due to check-read1, the \".*\" is matched to \"\" and the remaining\n$gdb_prompt  is read by the the following gdb_py_test_silent_cmd, which has\nits own $gdb_prompt read by the following gdb_py_test_silent_cmd, which has\nits own $gdb_prompt causing a mismatch for the following gdb_test_multiple:\n...\ngdb_test_multiple \"python gdb.prompt_hook = error_prompt\" \"set the hook\" {\n    -re \"Traceback.*File.*line.*RuntimeError.*Python exception called.*\" {\n        pass \"set hook\"\n    }\n}\n...\nwhich causes both FAILs.\n\nThe second gdb_test_multiple has the same problem as the first, but it happens\nnot to cause a FAIL because it's followed by a gdb_py_test_silent_cmd and a\nclean_restart.\n\nFix the regexps in both gdb_test_multiple calls.\n\nTested on x86_64-linux.\n\ngdb/testsuite/ChangeLog:\n\n2019-07-29  Tom de Vries  <tdevries@suse.de>\n\n\t* gdb.python/python.exp: Don't terminate gdb_test_multiple regexp\n\twith \".*\".",
    "tree": {
      "sha": "6f8733d0a8aaded97e7c846df1643e3b8acabdd7",
      "url": "https://api.github.com/repos/bminor/binutils-gdb/git/trees/6f8733d0a8aaded97e7c846df1643e3b8acabdd7"
    },
    "url": "https://api.github.com/repos/bminor/binutils-gdb/git/commits/5beafce944b6b200d1fa80d7f4a8656cb2ecf2fa",
    "comment_count": 0,
    "verification": {
      "verified": false,
      "reason": "unsigned",
      "signature": null,
      "payload": null
    }
  },
  "url": "https://api.github.com/repos/bminor/binutils-gdb/commits/5beafce944b6b200d1fa80d7f4a8656cb2ecf2fa",
  "html_url": "https://github.com/bminor/binutils-gdb/commit/5beafce944b6b200d1fa80d7f4a8656cb2ecf2fa",
  "comments_url": "https://api.github.com/repos/bminor/binutils-gdb/commits/5beafce944b6b200d1fa80d7f4a8656cb2ecf2fa/comments",
  "author": {
    "login": "vries",
    "id": 4057235,
    "node_id": "MDQ6VXNlcjQwNTcyMzU=",
    "avatar_url": "https://avatars.githubusercontent.com/u/4057235?v=4",
    "gravatar_id": "",
    "url": "https://api.github.com/users/vries",
    "html_url": "https://github.com/vries",
    "followers_url": "https://api.github.com/users/vries/followers",
    "following_url": "https://api.github.com/users/vries/following{/other_user}",
    "gists_url": "https://api.github.com/users/vries/gists{/gist_id}",
    "starred_url": "https://api.github.com/users/vries/starred{/owner}{/repo}",
    "subscriptions_url": "https://api.github.com/users/vries/subscriptions",
    "organizations_url": "https://api.github.com/users/vries/orgs",
    "repos_url": "https://api.github.com/users/vries/repos",
    "events_url": "https://api.github.com/users/vries/events{/privacy}",
    "received_events_url": "https://api.github.com/users/vries/received_events",
    "type": "User",
    "site_admin": false
  },
  "committer": {
    "login": "vries",
    "id": 4057235,
    "node_id": "MDQ6VXNlcjQwNTcyMzU=",
    "avatar_url": "https://avatars.githubusercontent.com/u/4057235?v=4",
    "gravatar_id": "",
    "url": "https://api.github.com/users/vries",
    "html_url": "https://github.com/vries",
    "followers_url": "https://api.github.com/users/vries/followers",
    "following_url": "https://api.github.com/users/vries/following{/other_user}",
    "gists_url": "https://api.github.com/users/vries/gists{/gist_id}",
    "starred_url": "https://api.github.com/users/vries/starred{/owner}{/repo}",
    "subscriptions_url": "https://api.github.com/users/vries/subscriptions",
    "organizations_url": "https://api.github.com/users/vries/orgs",
    "repos_url": "https://api.github.com/users/vries/repos",
    "events_url": "https://api.github.com/users/vries/events{/privacy}",
    "received_events_url": "https://api.github.com/users/vries/received_events",
    "type": "User",
    "site_admin": false
  },
  "parents": [
    {
      "sha": "2d274232f3e693d54f8dd561de73810d1a03ddf7",
      "url": "https://api.github.com/repos/bminor/binutils-gdb/commits/2d274232f3e693d54f8dd561de73810d1a03ddf7",
      "html_url": "https://github.com/bminor/binutils-gdb/commit/2d274232f3e693d54f8dd561de73810d1a03ddf7"
    }
  ],
  "stats": {
    "total": 9,
    "additions": 7,
    "deletions": 2
  },
  "files": [
    {
      "sha": "c2f9d6637b5427404c96d3958c2f430ebf59365d",
      "filename": "gdb/testsuite/ChangeLog",
      "status": "modified",
      "additions": 5,
      "deletions": 0,
      "changes": 5,
      "blob_url": "https://github.com/bminor/binutils-gdb/blob/5beafce944b6b200d1fa80d7f4a8656cb2ecf2fa/gdb/testsuite/ChangeLog",
      "raw_url": "https://github.com/bminor/binutils-gdb/raw/5beafce944b6b200d1fa80d7f4a8656cb2ecf2fa/gdb/testsuite/ChangeLog",
      "contents_url": "https://api.github.com/repos/bminor/binutils-gdb/contents/gdb/testsuite/ChangeLog?ref=5beafce944b6b200d1fa80d7f4a8656cb2ecf2fa",
      "patch": "@@ -1,3 +1,8 @@\n+2019-07-29  Tom de Vries  <tdevries@suse.de>\n+\n+\t* gdb.python/python.exp: Don't terminate gdb_test_multiple regexp\n+\twith \".*\".\n+\n 2019-07-29  Tom de Vries  <tdevries@suse.de>\n \n \t* lib/gdb.exp (skip_libstdcxx_probe_tests_prompt, gdb_is_target_1):"
    },
    {
      "sha": "fe55234060bc81faefb4d280d3699749fbe7cbcf",
      "filename": "gdb/testsuite/gdb.python/python.exp",
      "status": "modified",
      "additions": 2,
      "deletions": 2,
      "changes": 4,
      "blob_url": "https://github.com/bminor/binutils-gdb/blob/5beafce944b6b200d1fa80d7f4a8656cb2ecf2fa/gdb/testsuite/gdb.python/python.exp",
      "raw_url": "https://github.com/bminor/binutils-gdb/raw/5beafce944b6b200d1fa80d7f4a8656cb2ecf2fa/gdb/testsuite/gdb.python/python.exp",
      "contents_url": "https://api.github.com/repos/bminor/binutils-gdb/contents/gdb/testsuite/gdb.python/python.exp?ref=5beafce944b6b200d1fa80d7f4a8656cb2ecf2fa",
      "patch": "@@ -453,7 +453,7 @@ gdb_py_test_multiple \"prompt substitution readline\" \\\n   \"end\" \"\"\n \n gdb_test_multiple \"python gdb.prompt_hook = error_prompt\" \"set the hook\" {\n-    -re \"Python Exception (exceptions.RuntimeError|<(type 'exceptions.|class ')RuntimeError'>) Python exception called.*\" {\n+    -re \"Python Exception (exceptions.RuntimeError|<(type 'exceptions.|class ')RuntimeError'>) Python exception called.*$gdb_prompt $\" {\n \tpass \"set hook\"\n     }\n }\n@@ -465,7 +465,7 @@ gdb_py_test_silent_cmd \"set python print-stack full\" \\\n     \"set print-stack full for prompt error test\" 1\n \n gdb_test_multiple \"python gdb.prompt_hook = error_prompt\" \"set the hook\" {\n-    -re \"Traceback.*File.*line.*RuntimeError.*Python exception called.*\" {\n+    -re \"Traceback.*File.*line.*RuntimeError.*Python exception called.*$gdb_prompt $\" {\n \tpass \"set hook\"\n     }\n }"
    }
  ]
}